src/taipy/core/data/data_node.py

```diff
@@ -320,12 +320,17 @@ def write(self, data, job_id: Optional[JobId] = None, **kwargs: Dict[str, Any]):
         from ._data_manager_factory import _DataManagerFactory
 
         self._write(data)
-        self._track_edit(job_id=job_id, **kwargs)
+        self.track_edit(job_id=job_id, **kwargs)
         self.unlock_edit()
         _DataManagerFactory._build_manager()._set(self)
 
-    def _track_edit(self, **options):
-        """Add Edit tracking information to this data node."""
+    def track_edit(self, **options):
+        """Add edit tracking information to this data node.
+
+        Parameters:
+            options: track `timestamp`, `comments`, `job_id`. The others are user-custom, users can use options
```

Comment:
Added description for `options` parameter. Let me know what you think? @trgiangdo @jrobinAV

Reply:
I am really confused now.
Do we track track `timestamp`, `comments`, `job_id` by default, and the user can add more tracking options?
Or when the user provides new options, `timestamp`, `comments`, `job_id` will not be tracked anymore?

Reply:
We track timestamp, comments, job_id, and user can add more information that suit their case. On second thought. should we add timestamp, comments, job_id to the signature to make it explicit? Something like: 
`def track_edit(self,  timestamp: datetime.datetime = None, comments: str = None, job_id: JobId = None, **options):`

What do you think? @jrobinAV @trgiangdo

Reply:
I think we should clear it in the doc only.
The default value of timestamp, comments, job_id are None may cause confusion.

Reply:
hmm I don't think it will cause any confusions though :/

Reply:
First, we can improve the whole class documentation. not only the method documentation. We can add more context to this method, which is greatly linked to the `edits` attribute. We can explain what an edit is and how it is updated and used by Taipy. We can add an explanation for each of the three Taipy options. 

1. **timestamp** represents the time instant of the writing. It is used to sort the edits and trace a data node's change history. It is populated when the `write` method is called (by task execution or manually), when the data node is edited from the `data_node` viz elmt, or when the `track_edit` is called. It is displayed in the `data_node` visual element. 
2. **comments** represents a free text to explain or comment on a change of a data node. It can be populated by an end-user when the data node is edited from the `data_node` viz elmt, the `write` method is called manually. It is displayed in the `data_node` visual element.
3. **job_id** is only populated when the data node is written by a task execution and corresponds to the job's id. It is displayed in the `data_node` visual element.
4. We can give examples of custom options when the `write` and `track_edit` methods are manually called. The name of the third-party software. The id of the process that modified the data. The contact information of the IT responsible for the change, The user name of the end-user that edited the data, etc.

Once the `edits` attribute is clearly documented, we can say that the method creates and adds a new entry in the `edits` attribute without writing the data. We can refer to the edits attribute documentation and explain the motivation to open this API. It is particularly handy when data (referred by the data node) is modified independently from Taipy tasks and data nodes (a file is updated on the disk, a database is written by third-party software, etc.). 
It also lets Taipy know that some data node has been edited, so skippable downstream tasks should not be skipped.
We can also repeat the examples of custom options as well.

Reply:
First of all, that's more work for Toan 😂

Secondly, I'm not sure about the example, since this is in refman and I don't want it to be too long.
But a short and simple one is better than none I guess.

Reply:
The ref manual can be more detailed than what we used to do. For instance, the GUI team uses much more the ref manual than we do.
There is no need to add all the details for all Classes and functions. However, this feature is tricky. I believe it does make sense.